### ISSUE TITLE
Tracking only works when injecting old GA code

### ISSUE THREAD

**Issue by lwensveen** [luisfarzati/angulartics#321]
Hello,

I'm running into some problems when implementing Google tag manager with universal analytics.

I've added all the Macros, Rules and Events as described here: https://github.com/luisfarzati/angulartics

When I inject my old tracking code like this:

    <script>
     (function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
     (i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
     m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
      })(window,document,'script','//www.google-analytics.com/analytics.js','ga');

     ga('create', 'UA-XXXXXXXXX', 'auto');

    </script>

All of a sudden my tracking starts to work.

I reckon this should've been handled by Angulartics, am I correct?

**Comment by timharrison2015** [comment]
If you're using Google Tag Manager you don't need an analytics tracker on the page, you can just create your normal page view tracking in tag manager. 
Then you can add a don't fire rule for the URL that you have Angulartics on.
There is a tracking ID field in Angulartics for your UA account number so I presume that this data gets pushed to your tracking code for data collection.

(From comments elsewhere on this thread)

app.config('googleAnalyticsCordovaProvider', function(googleAnalyticsCordovaProvider) {
googleAnalyticsCordovaProvider.trackingId = 'XXXXX';
googleAnalyticsCordovaProvider.period = 20; // default: 10 (in seconds)
googleAnalyticsCordovaProvider.debug = false; // default: false
});

**Comment by jeroenbr** [comment]
Same problem here. We're currently injecting the analytics tracking code with GTM. If not injecting the snippet, there's no traffic visible in Analytics. 

The events are sent and everything seems to work in GTM. We don't need to use 
ga('create', 'UA-XXXXXXXXX', 'auto') in the injected code because the UA id is already set in GTM, so there is some magic Google is doing with linking up GTM with analytics. 

Basically, without (//www.google-analytics.com/)analytics.js loaded, we see no traffic.

**Comment by DanWilkerson** [comment]
jeroeonbr, do you have a pageview tag setup in Google Tag Manager? This should be a tag of the Google Analytics type that uses Universal Analytics and the track type Pageview. If you do, let me know what version of GTM you are using and I'll see if I can help.

**Comment by lwensveen** [comment]
Hey Dan,

Thanks for the help, I've done exactly that, as far as I know.
I'm using v1 by the way.

This is how I've configured it:

![screen shot 2015-05-10 at 23 47 25](https://cloud.githubusercontent.com/assets/4800388/7556414/f34d9e56-f76e-11e4-9680-679a1c0a316d.png)

I reckon this should be allright?

**Comment by DanWilkerson** [comment]
@lwensveen Hm, potentially dumb question here, but is thay Tracking ID what you have in prod or did you obfuscate it for the screenshot? When you use the Debug tool, do you see the tags being fired as you navigate the app? You can also PM me a link and I'll take a look.

**Comment by lwensveen** [comment]
@DanWilkerson No it's not, I edited it so it wasn't there to see for the whole world. When I use the debug tool, everything works fine. I see the events being fired, but they just don't show up in the analytics dashboard, until I add the script tag I've mentioned in my topic start.

**Comment by hermanocabral** [comment]
I'm having the same issue btw, @lwensveen did you managed to get anywhere with this?

**Comment by lwensveen** [comment]
@hermanocabral I've been trying but I haven't been successful so far, did you have any luck?

Maybe @DanWilkerson has some more info he could share?

**Comment by DanWilkerson** [comment]
@hermanocabral @lwensveen I'm happy to help, but there could be several
configuration issues at play. It would be easiest for me if you could DM me
a link where I could inspect the app itself and check for some common
issues.

If the Data Layer event is firing and the Google Analytics tag is firing,
it could be that the UA number is incorrect or missing, you could be
filtering yourself out, or your Realtime Analytics might not be warmed up
yet. If the tag isn't firing, your Trigger may be misconfigured (see #352).
Again, it's probably quicker to send me a link and I can take a peek.

@hermanocabral <https://github.com/hermanocabral> I've been trying but I
haven't been successful so far, did you have any luck?

Maybe @DanWilkerson <https://github.com/DanWilkerson> has some more info he
could share?

—
Reply to this email directly or view it on GitHub
<https://github.com/luisfarzati/angulartics/issues/321#issuecomment-103033605>
.

**Comment by hermanocabral** [comment]
I managed to get the new analytics tag working with a "hack" but honestly,  I didn't spend much time on it.

I'm using Google Tag Manager (GTM) to inject Google Analytics (GA) into my app. What I did to get it working with angulartics was simply call ```ga('create', ....);``` during my app's initialization, passing in the tracking ID from GTM.

I've only spent like 5 minutes testing this but apparently the issue is related to **when** the GA global obj is created/initialized in relation to the angular obj. As long as angulartics is initialized first, GA works fine.

**Comment by timelf123** [comment]
Can anyone reproduce this issue?

> I've only spent like 5 minutes testing this but apparently the issue is related to when the GA global obj is created/initialized in relation to the angular obj. As long as angulartics is initialized first, GA works fine.

**Comment by luisfarzati** [comment]
Closing for now -- please feel free to reopen if the issue persists.
